### PR TITLE
[QT-687] Rewrite packaging action

### PR DESCRIPTION
**Pull request by ryancragun**
This is a proposal for a possible v2 of this action. It makes the following changes

* Rewrite to run as a composite action instead of Docker action. We do this for two primary reasons:
  * Building the action for every packaging workflow is slow. Depending on the runner this could be anywhere from 15 seconds to a minute. This is work that we need not do on every packaging workflow invocation.
  * It relies on the Docker hub which often throws 429 errors in packaging workflows.
* Use upstream build artifacts instead of building nFPM from source
  * We still require Github with this method but instead of compiling the binary from source we install the pre-built artifact.
  * We no longer depends on Kyle's branch which was already a bit of a security risk since it runs unpinned code from an external repository that is pointing at the binary we're packing. (The bug fix in his branch was fixed upstream already [0])

Because of the new change in implementation we also consider:

* Whether or not to install Go for the template generator. This allows callers to set up their own Go toolchain if they want.
* Add testing for this Go behavior.
* Add new inputs for nFPM version and installation path.

If we can assume that this workflow was only ever run on Linux or macOS runners then it should be fully backwards compatible. If not, it will not work on windows at this time.

[0] https://github.com/goreleaser/nfpm/blob/main/deb/deb_test.go#L217

### CODE REVIEW

File: action.yml [hunk 1]
```diff
@@ -80,10 +80,102 @@ inputs:
...
+    INPUT_ARCH: ${{ inputs.arch }}
+    INPUT_VERSION: ${{ inputs.version }}
+    INPUT_MAINTAINER: ${{ inputs.maintainer }}
+    INPUT_VENDOR: ${{ inputs.vendor }}
+    INPUT_DESCRIPTION: ${{ inputs.description }}
+    INPUT_HOMEPAGE: ${{ inputs.homepage
+    INPUT_LICENSE: ${{ inputs.license }}
+    INPUT_DEPENDS: ${{ inputs.depends }}
+    INPUT_BINARY: ${{ inputs.binary }}
+    INPUT_BIN_PATH: ${{ inputs.bin_path }}
```
Comment (jeanneryan) [inline action.yml:105]: What is the `INPUT_BIN_PATH`? I'm comparing it to the list of inputs from the README and this looks like it's new.

Comment (ryancragun) [reply action.yml:105]: I copied it directly from the previous docker vars and missed that it was no longer an input.

**Review by jeanneryan — APPROVED** [review]

**Comment by shore** [comment]
> If we can assume that this workflow was only ever run on Linux or macOS runners then it should be fully backwards compatible. If not, it will not work on windows at this time.

A quick(ish) GitHub search only shows it being used from these contexts, I think we're good on that front and don't need a major version bump.

**Review by shore — APPROVED** [review]
Thank you, Ryan!

**Comment by ryancragun** [comment]
I cleaned it up and tested in my fork: https://github.com/ryancragun/actions-packaging-linux/actions/runs/8288284520

File: action.yml (hunk 1 above)
Comment (ryancragun) [reply action.yml]: I was actually wrong, it's used by the template generator to set the package destination. I fixed it in my latest commit.

**Comment by ryancragun** [comment]
@shore / @jeanneryan Any chance we can get this merged (but not tagged) so I can start testing it in Vault? As long we don't update the tag I wouldn't expect it to have much downstream impact unless people are using `@main`

**Comment by shore** [comment]
> @shore / @jeanneryan Any chance we can get this merged (but not tagged) so I can start testing it in Vault? As long we don't update the tag I wouldn't expect it to have much downstream impact unless people are using `@main`

done!

**Pull request merged**